src/draw/lv_draw_vector.h

```diff
@@ -0,0 +1,213 @@
+    lv_opa_t opa;
+    lv_vpath_fill_t fill_rule;
+    lv_vpath_pattern_t pattern_mode;
+    lv_draw_img_dsc_t img_dsc;
+    lv_grad_dsc_t grad;
+    lv_matrix_t matrix;
+} lv_vpath_fill_dsc_t;
+
+typedef struct {
+    lv_color_t color;
+    lv_opa_t opa;
+    float width;
+    float * dash_pattern;            /* The array of consecutive pair values of the dash length and the gap length */
+    uint16_t dash_pattern_count;     /* The length of the dash_pattern array */
+    float dash_offset;               /* The shift of the starting point within the repeating dash pattern */
+    lv_vpath_stroke_cap_t cap;       /* The shape to be used at the end of open subpaths */
+    lv_vpath_stroke_join_t join;     /* The shape to be used at the corners of paths when they are joined */
+    uint16_t miter_limit;            /* A limit on the extent of the stroke join, the default value is 4 */
+    lv_grad_dsc_t grad;
+} lv_vpath_stroke_dsc_t;
+
+typedef struct {
+    uint8_t * path;
+    uint32_t path_length;
+    uint32_t memory_size;
+} lv_vpath_t;
+
+typedef struct {
+    lv_vpath_fill_dsc_t fill_dsc;
+    lv_vpath_stroke_dsc_t stroke_dsc;
+    lv_matrix_t matrix;
+    lv_vpath_blend_t blend;
+    lv_area_t scissor_area;
+    bool stroke_first;
+} lv_vpath_dsc_t;
+
+typedef struct lv_vpath_task_s {
+    lv_vpath_dsc_t dsc;
+    lv_vpath_t path;
+    struct lv_vpath_task_s * next;          /*Tasks need to be freed when finished*/
+} lv_vpath_task_t;
+
+typedef struct {
+    float x;
+    float y;
+} lv_fpoint_t;
+
+/**********************
+ * GLOBAL PROTOTYPES
+ **********************/
+
+/*Path operations, implemented by different engine*/
+lv_vpath_t * lv_vpath_create(void);
+lv_vpath_t * lv_vpath_copy(lv_vpath_t * path);
+void lv_vpath_destroy(lv_vpath_t * path);
+void lv_vpath_move_to(lv_vpath_t * path, lv_fpoint_t p);
+void lv_vpath_line_to(lv_vpath_t * path, lv_fpoint_t p);
+void lv_vpath_quad_to(lv_vpath_t * path, lv_fpoint_t p1, lv_fpoint_t p2);
+void lv_vpath_cubic_to(lv_vpath_t * path, lv_fpoint_t p1, lv_fpoint_t p2, lv_fpoint_t p3);
+void lv_vpath_arc_to(lv_vpath_t * path, lv_fpoint_t p1, lv_fpoint_t p2, float r);
+void lv_vpath_close(lv_vpath_t * path);
+void lv_vpath_append_rect(lv_vpath_t * path, lv_area_t * rect, float r);
+void lv_vpath_append_circle(lv_vpath_t * path, lv_fpoint_t c, float r);
+/**
+ * @param[in] start_angle The start angle of the arc given in degrees, measured counter-clockwise from the horizontal line.
+ * @param[in] sweep The central angle of the arc given in degrees, measured counter-clockwise from @p start_angle.
+ * @param[in] pie Specifies whether to draw radii from the arc's center to both of its end-point - drawn if @c true.
+ */
+void lv_vpath_append_arc(lv_vpath_t * path, lv_fpoint_t c, float r, uint16_t start_angle, uint16_t sweep,
+                         bool pie);
+void lv_vpath_append_path(lv_vpath_t * path, lv_vpath_t * subpath);
+
+/*Param operations*/
+void lv_vpath_dsc_init(lv_vpath_dsc_t * dsc);
```

Comment:
```suggestion
void lv_vpath_dsc_init(lv_vpath_dsc_t * dsc);
```
In order to save the **lv_vpath_task_t** we can change **lv_vpath_dsc_t** to **lv_vpath_draw_context**
That we can create and maintain it using methods like:
```
lv_vpath_draw_context*  lv_vpath_draw_context_create(struct _lv_layer_t * layer);
lv_vpath_draw_context_destroy(lv_vpath_draw_context* ctx)
```
The drawing function can be:
```
void lv_vpath_draw( lv_vpath_draw_context * ctx, lv_vpath_t * path);
```

Reply:
@PeterBee97  Do you think this is better?